src/Windows/R/Editor/Impl/Settings/REditorSettings.cs

```diff
@@ -24,15 +26,28 @@ public sealed class REditorSettings: EditorSettings, IWritableREditorSettings, I
 
         private RFormatOptions _formatOptions = new RFormatOptions();
 
-        public REditorSettings(ICoreShell coreShell): base(coreShell, RContentTypeDefinition.ContentType) { }
-        public REditorSettings(IEditorSettingsStorage storage) : base(storage) { }
+        public REditorSettings(ICoreShell coreShell) : base(coreShell, RContentTypeDefinition.ContentType) {
+            CreateLintOptions();
+        }
+
+        public REditorSettings(IEditorSettingsStorage storage) : base(storage) {
+            CreateLintOptions();
+        }
 
         public override void ResetSettings() {
             _formatOptions = new RFormatOptions();
-            LintOptions = new LintOptions();
+            CreateLintOptions();
             base.ResetSettings();
         }
 
+        private void CreateLintOptions() {
+            LintOptions = new LintOptions();
+            LintOptions.PropertyChanged += _OnLintOptionsPropertyChanged;
```

Comment:
Ok, first of all, I think I've found a bug in `SettingsHolder`:

```cs
            public T GetValue<T>(T defaultValue, [CallerMemberName] string name = null) {
                object value;
                return _dict.TryGetValue(name, out value) ? (T)value : default(T);
            }
```

`defaultValue` isn't used, it should probably be 

```cs
            public T GetValue<T>(T defaultValue, [CallerMemberName] string name = null) 
                => _dict.TryGetValue(name, out object value) ? (T)value : defaultValue;
```